synapse/handlers/e2e_keys.py

```diff
@@ -981,8 +989,48 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
             user_id, key_type, from_user_id
         )
         if key is None:
-            logger.debug("no %s key found for %s", key_type, user_id)
+            # Attempt to fetch the missing key from the remote user's server
+            user = UserID.from_string(user_id)
+            try:
+                remote_result = yield self.federation.query_client_keys(
```

Comment:
Should check if the user is a remote user, and only do the federation query if so.